### PR TITLE
 move webhook admission to generic apiserver

### PR DESCRIPTION
**Pull request by deads2k**
This builds on https://github.com/kubernetes/kubernetes/pull/54414.

It moves the admission webhook into the generic apiserver.  It also allows the injection of a server managed way to override the rest.Config used for a particular service.  This makes for an easier point to allow auto-configuration of the kube-apiserver to loopback to itself as a special case for kubernetes.default.svc.

@kubernetes/sig-api-machinery-pr-reviews 

```release-note
the generic admission webhook is now available in the generic apiserver
```

### CODE REVIEW

**Comment by lavalamp** [comment]
Didn't get to this today, will review tomorrow.

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/webhook/admission.go [hunk 1]
```diff
@@ -40,10 +41,11 @@ import (
...
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	admissioninit "k8s.io/kubernetes/pkg/kubeapiserver/admission"
+)
 
-	// install the clientgo admission API for use with api registry
-	_ "k8s.io/kubernetes/pkg/apis/admission/install"
+const (
+	// Name of admission plug-in
+	PluginName = "GenericAdmissionWebhook"
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/webhook/admission.go]: nit: just WebHook?

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/webhook/admission.go]: ok, that name is not new. So we have to stick. Please change the name in `Register` below to use the constant.

File: cmd/kube-apiserver/app/server.go [hunk 2]
```diff
@@ -481,7 +492,7 @@ func BuildGenericConfig(s *options.ServerRunOptions, proxyTransport *http.Transp
 }
 
 // BuildAdmissionPluginInitializer constructs the admission plugin initializer
-func BuildAdmissionPluginInitializer(s *options.ServerRunOptions, client internalclientset.Interface, sharedInformers informers.SharedInformerFactory, serviceResolver aggregatorapiserver.ServiceResolver) (admission.PluginInitializer, error) {
+func BuildAdmissionPluginInitializer(s *options.ServerRunOptions, client internalclientset.Interface, sharedInformers informers.SharedInformerFactory, serviceResolver aggregatorapiserver.ServiceResolver, webhookAuthWrapper webhook.AuthenticationInfoResolverWrapper) (admission.PluginInitializer, error) {
```
Comment (lavalamp) [inline cmd/kube-apiserver/app/server.go]: We should make a rule about how many parameters we're allowed to make before a config struct is required--I think this function (and many in this file) is over the threshold. ;)

**Comment by lavalamp** [comment]
lgtm when tests pass

**Comment by deads2k** [comment]
It's flaking on https://github.com/kubernetes/kubernetes/issues/54619.  I've opened a fix.

**Comment by deads2k** [comment]
tagging per comment.  Daniel is top level, so approving too.

**Pull request merged**